src/App.tsx

```diff
@@ -1,60 +1,141 @@
-import classNames from 'classnames';
-
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
 import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
+import { getUsers } from './services/user';
 import { UserSelector } from './components/UserSelector';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getPosts } from './services/post';
 import { Loader } from './components/Loader';
+import classNames from 'classnames';
+import { PostDetails } from './components/PostDetails';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[] | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isOpenedNewCommentForm, setIsOpenedNewCommentForm] = useState(false);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const showNoPost =
+    selectedUser && !isLoading && error.length === 0 && posts?.length === 0;
+  const showPostList =
+    selectedUser &&
+    !isLoading &&
+    error.length === 0 &&
+    posts &&
+    posts.length > 0;
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(usersLoadingError => {
+        setError('usersLoadingError');
```

Comment:
The error message 'usersLoadingError' is being set as a string. Consider using a more descriptive error message or handling the error object to provide more context to the user.